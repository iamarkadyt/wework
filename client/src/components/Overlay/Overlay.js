import React from 'react'
import './Overlay.scss'

class Overlay extends React.Component {
  componentWillMount() {
    document.body.style = "overflow: hidden"
  }

  componentWillUnmount() {
    document.body.style = null
  }

  handleSubmit(e) {
    const { 
      onBackdropClick 
    } = this.props

    if (e.target.id === 'Overlay-container-backdrop') {
      onBackdropClick()
    }
  }

  render() {
    const { 
      children, 
      centered
    } = this.props

    return (
      <div id="Overlay-container-backdrop" className="Overlay-container" onClick={this.handleSubmit}>
        <div className={["Overlay-content", centered ? "Overlay-content-centered" : null].join(' ')}>
          {children}
        </div>
      </div>
    )
  }
}

export default Overlay
