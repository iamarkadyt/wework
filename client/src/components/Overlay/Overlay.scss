@import 'variables.scss';
.Overlay-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    min-height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: calc(#{$navigation_height} + 1rem) 0 2rem 0;
    .content {
        z-index: 1;
        background: white;
        padding: 0.3rem;
        border-radius: .3rem;
        box-shadow: $box_shadow_main;
    }
    .backdrop {
        position: fixed;
        top: 0;
        left: 0;
        width: 100vw;
        height: 100vh;
        background-color: rgba(0, 0, 0, 0.5);
    }
}