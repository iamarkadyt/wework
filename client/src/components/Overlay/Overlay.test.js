import React from 'react'
import { shallow } from 'enzyme'
import Overlay from './Overlay'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    children: 'children',
    centered: true,
    onBackdropClick: jest.fn()
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Overlay {...props} />)
  }
  return mountedComponent
}

describe('Overlay', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('props', () => {
    describe('effects', () => {
      describe('centered', () => {
        it('if true, content div receives additional class', () => {
          props.centered = true
          expect(comp().find('.Overlay-content').hasClass('Overlay-content-centered')).toBe(true)
        })

        it('if false, content div does not receive any additional classes', () => {
          props.centered = false
          expect(comp().find('.Overlay-content').prop('className').trim()).toBe('Overlay-content')
        })
      })
    })

    describe('passing', () => {
      it('outer div receives correct function into onClick', () => {
        expect(comp().find('.Overlay-container').prop('onClick')).toBe(comp().instance().handleSubmit)
      })
    })
  })

  describe('functions', () => {
    describe('handleSubmit', () => {
      it('calls onBackdropClick if e.target.id is a backdrop div', () => {
        props.onBackdropClick = jest.fn()
        const e = { target: { id: 'Overlay-container-backdrop' } }
        comp().instance().handleSubmit(e)
        expect(props.onBackdropClick).toHaveBeenCalled()
      })

      it('does NOT call onBackdropClick if e.target.id is NOT a backdrop div', () => {
        props.onBackdropClick = jest.fn()
        const e = { target: { id: 'something-else' } }
        comp().instance().handleSubmit(e)
        expect(props.onBackdropClick).not.toHaveBeenCalled()
      })
    })

    describe('componentWillMount', () => {
      it('sets specific document.body.style', () => {
        comp()
        expect(global.document.body.style._values.overflow).toEqual('hidden')
      })
    })

    describe('componentWillUnmount', () => {
      it('removes added styles', () => {
        comp().unmount()
        expect(Object.keys(global.document.body.style._values)).toHaveLength(0)
      })
    })
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})


