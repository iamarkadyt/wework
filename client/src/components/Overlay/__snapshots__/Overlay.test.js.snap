// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay matches snapshot 1`] = `
<div
  className="Overlay-container"
  id="Overlay-container-backdrop"
  onClick={[Function]}
>
  <div
    className="Overlay-content Overlay-content-centered"
  >
    children
  </div>
</div>
`;
